llvm/lib/Passes/PassBuilderPipelines.cpp

```diff
@@ -1615,6 +1617,9 @@ PassBuilder::buildModuleOptimizationPipeline(OptimizationLevel Level,
   MPM.addPass(createModuleToFunctionPassAdaptor(std::move(OptimizePM),
                                                 PTO.EagerlyInvalidateAnalyses));
 
+  if (LTOPostLink)
```

Comment:
Maybe add a comment that for non-LTO compiles this is added to the pipeline by Clang? I suppose you could just enable it here always and remove it from the Clang setup, up to you.

Also, maybe add a comment here that it is done late so that other call transformations, e.g. those done by InstCombine, are complete.

ditto for the main invocation in the full LTO default pipeline.